{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(82188, 160)"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv('../data/data_nc.txt', sep='\\t')\n",
    "data['#Chrom'] = data['#Chrom'].apply(lambda x: str(x))\n",
    "data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    81140\n",
       "True       430\n",
       "Name: oldTestset, dtype: int64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.oldTestset.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    81536\n",
       "True       652\n",
       "Name: NCBoost_data, dtype: int64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.NCBoost_data.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Benign        79577\n",
       "Pathogenic     2611\n",
       "Name: label, dtype: int64"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.label.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "//anaconda3/envs/py3_vp/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "cadd = pd.read_csv('../data/data_nc.ncboost_cadd.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>annovar_annotation</th>\n",
       "      <th>closest_gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>pLI</th>\n",
       "      <th>familyMemberCount</th>\n",
       "      <th>ncRVIS</th>\n",
       "      <th>...</th>\n",
       "      <th>CpG</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>bStatistic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>980669</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.040303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.319</td>\n",
       "      <td>-0.257</td>\n",
       "      <td>-0.395</td>\n",
       "      <td>5.15</td>\n",
       "      <td>-2.60</td>\n",
       "      <td>850.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>980948</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.040303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.455</td>\n",
       "      <td>-0.308</td>\n",
       "      <td>0.301</td>\n",
       "      <td>3.51</td>\n",
       "      <td>-1.13</td>\n",
       "      <td>849.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>981087</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.040303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>-1.215</td>\n",
       "      <td>-1.268</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-6.13</td>\n",
       "      <td>849.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>981328</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.040303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.144</td>\n",
       "      <td>-0.637</td>\n",
       "      <td>-0.842</td>\n",
       "      <td>4.07</td>\n",
       "      <td>-3.17</td>\n",
       "      <td>849.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>982356</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.040303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.550</td>\n",
       "      <td>-0.527</td>\n",
       "      <td>-0.137</td>\n",
       "      <td>3.98</td>\n",
       "      <td>-2.43</td>\n",
       "      <td>850.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42397</td>\n",
       "      <td>X</td>\n",
       "      <td>154489806</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.694532</td>\n",
       "      <td>83.0</td>\n",
       "      <td>-0.663738</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42398</td>\n",
       "      <td>X</td>\n",
       "      <td>154507173</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>CLIC2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.450380</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.645853</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42399</td>\n",
       "      <td>Y</td>\n",
       "      <td>2655017</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SRY</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.428870</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42400</td>\n",
       "      <td>Y</td>\n",
       "      <td>2655719</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SRY</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.428870</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42401</td>\n",
       "      <td>Y</td>\n",
       "      <td>2655774</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>SRY</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.428870</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.108606</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42402 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos ref alt annovar_annotation closest_gene_name  \\\n",
       "0       1     980669   A   G           intronic              AGRN   \n",
       "1       1     980948   C   T           intronic              AGRN   \n",
       "2       1     981087   A   G           intronic              AGRN   \n",
       "3       1     981328   C   G           intronic              AGRN   \n",
       "4       1     982356   G   A           intronic              AGRN   \n",
       "...    ..        ...  ..  ..                ...               ...   \n",
       "42397   X  154489806   C   A               UTR3            RAB39B   \n",
       "42398   X  154507173   T   G               UTR3             CLIC2   \n",
       "42399   Y    2655017   T   C               UTR3               SRY   \n",
       "42400   Y    2655719   C   T               UTR5               SRY   \n",
       "42401   Y    2655774   C   G               UTR5               SRY   \n",
       "\n",
       "            gene_type       pLI  familyMemberCount    ncRVIS  ...   CpG  \\\n",
       "0      protein_coding  0.040303                0.0 -0.108606  ...  0.12   \n",
       "1      protein_coding  0.040303                0.0 -0.108606  ...  0.07   \n",
       "2      protein_coding  0.040303                0.0 -0.108606  ...  0.09   \n",
       "3      protein_coding  0.040303                0.0 -0.108606  ...  0.16   \n",
       "4      protein_coding  0.040303                0.0 -0.108606  ...  0.13   \n",
       "...               ...       ...                ...       ...  ...   ...   \n",
       "42397  protein_coding  0.694532               83.0 -0.663738  ...   NaN   \n",
       "42398  protein_coding  0.450380                9.0 -0.645853  ...   NaN   \n",
       "42399  protein_coding  0.428870                0.0 -0.108606  ...   NaN   \n",
       "42400  protein_coding  0.428870                0.0 -0.108606  ...   NaN   \n",
       "42401  protein_coding  0.428870                0.0 -0.108606  ...   NaN   \n",
       "\n",
       "       priPhCons  mamPhCons  verPhCons  priPhyloP  mamPhyloP  verPhyloP  \\\n",
       "0          0.000        0.0        0.0     -0.319     -0.257     -0.395   \n",
       "1          0.001        0.0        0.0     -0.455     -0.308      0.301   \n",
       "2          0.000        0.0        0.0     -0.333     -1.215     -1.268   \n",
       "3          0.000        0.0        0.0     -0.144     -0.637     -0.842   \n",
       "4          0.002        0.0        0.0      0.550     -0.527     -0.137   \n",
       "...          ...        ...        ...        ...        ...        ...   \n",
       "42397        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "42398        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "42399        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "42400        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "42401        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "\n",
       "       GerpN  GerpS  bStatistic  \n",
       "0       5.15  -2.60       850.0  \n",
       "1       3.51  -1.13       849.0  \n",
       "2       4.09  -6.13       849.0  \n",
       "3       4.07  -3.17       849.0  \n",
       "4       3.98  -2.43       850.0  \n",
       "...      ...    ...         ...  \n",
       "42397    NaN    NaN         NaN  \n",
       "42398    NaN    NaN         NaN  \n",
       "42399    NaN    NaN         NaN  \n",
       "42400    NaN    NaN         NaN  \n",
       "42401    NaN    NaN         NaN  \n",
       "\n",
       "[42402 rows x 32 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "cadd['chr'] = cadd['chr'].apply(lambda x: str(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(27835, 48)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = cadd.merge(data[['#Chrom', 'Pos', 'Ref', 'Alt', 'GeneName', \"GC\", \"CpG\", \"priPhCons\", \"mamPhCons\", \"verPhCons\", \"priPhyloP\", \"mamPhyloP\", \"verPhyloP\", \"GerpN\", \"GerpS\", \"bStatistic\"]], how='inner', right_on=['#Chrom', 'Pos', 'Ref', 'Alt', 'GeneName'], left_on=['chr', 'pos', 'ref', 'alt', 'closest_gene_name'], suffixes=('_CADD-v1.3', '_CADD-v1.4'))\n",
    "merged.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr',\n",
       " 'pos',\n",
       " 'ref',\n",
       " 'alt',\n",
       " 'annovar_annotation',\n",
       " 'closest_gene_name',\n",
       " 'gene_type',\n",
       " 'pLI',\n",
       " 'familyMemberCount',\n",
       " 'ncRVIS',\n",
       " 'ncGERP',\n",
       " 'RVIS_percentile',\n",
       " 'slr_dnds',\n",
       " 'GDI',\n",
       " 'gene_age',\n",
       " 'UTR3',\n",
       " 'UTR5',\n",
       " 'downstream',\n",
       " 'intergenic',\n",
       " 'intronic',\n",
       " 'upstream',\n",
       " 'GC_CADD-v1.3',\n",
       " 'CpG_CADD-v1.3',\n",
       " 'priPhCons_CADD-v1.3',\n",
       " 'mamPhCons_CADD-v1.3',\n",
       " 'verPhCons_CADD-v1.3',\n",
       " 'priPhyloP_CADD-v1.3',\n",
       " 'mamPhyloP_CADD-v1.3',\n",
       " 'verPhyloP_CADD-v1.3',\n",
       " 'GerpN_CADD-v1.3',\n",
       " 'GerpS_CADD-v1.3',\n",
       " 'bStatistic_CADD-v1.3',\n",
       " '#Chrom',\n",
       " 'Pos',\n",
       " 'Ref',\n",
       " 'Alt',\n",
       " 'GeneName',\n",
       " 'GC_CADD-v1.4',\n",
       " 'CpG_CADD-v1.4',\n",
       " 'priPhCons_CADD-v1.4',\n",
       " 'mamPhCons_CADD-v1.4',\n",
       " 'verPhCons_CADD-v1.4',\n",
       " 'priPhyloP_CADD-v1.4',\n",
       " 'mamPhyloP_CADD-v1.4',\n",
       " 'verPhyloP_CADD-v1.4',\n",
       " 'GerpN_CADD-v1.4',\n",
       " 'GerpS_CADD-v1.4',\n",
       " 'bStatistic_CADD-v1.4']"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged = merged.reindex(sorted(merged.columns), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Alt</th>\n",
       "      <th>CpG_CADD-v1.3</th>\n",
       "      <th>CpG_CADD-v1.4</th>\n",
       "      <th>GC_CADD-v1.3</th>\n",
       "      <th>GC_CADD-v1.4</th>\n",
       "      <th>GDI</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>GerpN_CADD-v1.3</th>\n",
       "      <th>GerpN_CADD-v1.4</th>\n",
       "      <th>...</th>\n",
       "      <th>priPhCons_CADD-v1.4</th>\n",
       "      <th>priPhyloP_CADD-v1.3</th>\n",
       "      <th>priPhyloP_CADD-v1.4</th>\n",
       "      <th>ref</th>\n",
       "      <th>slr_dnds</th>\n",
       "      <th>upstream</th>\n",
       "      <th>verPhCons_CADD-v1.3</th>\n",
       "      <th>verPhCons_CADD-v1.4</th>\n",
       "      <th>verPhyloP_CADD-v1.3</th>\n",
       "      <th>verPhyloP_CADD-v1.4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.120000</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.701987</td>\n",
       "      <td>1338.60140</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>5.15</td>\n",
       "      <td>5.15</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.319</td>\n",
       "      <td>-0.319</td>\n",
       "      <td>A</td>\n",
       "      <td>0.10090</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.395</td>\n",
       "      <td>-0.395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.662252</td>\n",
       "      <td>1338.60140</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>3.51</td>\n",
       "      <td>3.51</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001</td>\n",
       "      <td>-0.455</td>\n",
       "      <td>-0.455</td>\n",
       "      <td>C</td>\n",
       "      <td>0.10090</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.301</td>\n",
       "      <td>0.301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.093333</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.655629</td>\n",
       "      <td>1338.60140</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>4.09</td>\n",
       "      <td>4.09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>A</td>\n",
       "      <td>0.10090</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.268</td>\n",
       "      <td>-1.268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.160000</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.688742</td>\n",
       "      <td>1338.60140</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>4.07</td>\n",
       "      <td>4.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.144</td>\n",
       "      <td>-0.144</td>\n",
       "      <td>C</td>\n",
       "      <td>0.10090</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.842</td>\n",
       "      <td>-0.842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.708609</td>\n",
       "      <td>1338.60140</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>3.98</td>\n",
       "      <td>3.98</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.550</td>\n",
       "      <td>0.550</td>\n",
       "      <td>G</td>\n",
       "      <td>0.10090</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.137</td>\n",
       "      <td>-0.137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27830</td>\n",
       "      <td>X</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.026667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.410596</td>\n",
       "      <td>198.58178</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.45</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.397</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27831</td>\n",
       "      <td>X</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.516556</td>\n",
       "      <td>198.58178</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.66</td>\n",
       "      <td>...</td>\n",
       "      <td>0.197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.509</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27832</td>\n",
       "      <td>X</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.311258</td>\n",
       "      <td>13.42332</td>\n",
       "      <td>CLIC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.76</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.439</td>\n",
       "      <td>T</td>\n",
       "      <td>0.07122</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27833</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.053333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.397351</td>\n",
       "      <td>0.82297</td>\n",
       "      <td>SRY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.12810</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27834</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.040000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.430464</td>\n",
       "      <td>0.82297</td>\n",
       "      <td>SRY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.12810</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.205</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.509</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27835 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom Alt  CpG_CADD-v1.3  CpG_CADD-v1.4  GC_CADD-v1.3  GC_CADD-v1.4  \\\n",
       "0          1   G           0.12       0.120000          0.70      0.701987   \n",
       "1          1   T           0.07       0.066667          0.66      0.662252   \n",
       "2          1   G           0.09       0.093333          0.66      0.655629   \n",
       "3          1   G           0.16       0.160000          0.69      0.688742   \n",
       "4          1   A           0.13       0.133333          0.71      0.708609   \n",
       "...      ...  ..            ...            ...           ...           ...   \n",
       "27830      X   C            NaN       0.026667           NaN      0.410596   \n",
       "27831      X   A            NaN       0.000000           NaN      0.516556   \n",
       "27832      X   G            NaN       0.013333           NaN      0.311258   \n",
       "27833      Y   T            NaN       0.053333           NaN      0.397351   \n",
       "27834      Y   G            NaN       0.040000           NaN      0.430464   \n",
       "\n",
       "              GDI GeneName  GerpN_CADD-v1.3  GerpN_CADD-v1.4  ...  \\\n",
       "0      1338.60140     AGRN             5.15             5.15  ...   \n",
       "1      1338.60140     AGRN             3.51             3.51  ...   \n",
       "2      1338.60140     AGRN             4.09             4.09  ...   \n",
       "3      1338.60140     AGRN             4.07             4.07  ...   \n",
       "4      1338.60140     AGRN             3.98             3.98  ...   \n",
       "...           ...      ...              ...              ...  ...   \n",
       "27830   198.58178   RAB39B              NaN             3.45  ...   \n",
       "27831   198.58178   RAB39B              NaN             4.66  ...   \n",
       "27832    13.42332    CLIC2              NaN             4.76  ...   \n",
       "27833     0.82297      SRY              NaN             0.00  ...   \n",
       "27834     0.82297      SRY              NaN             0.00  ...   \n",
       "\n",
       "       priPhCons_CADD-v1.4  priPhyloP_CADD-v1.3  priPhyloP_CADD-v1.4  ref  \\\n",
       "0                    0.000               -0.319               -0.319    A   \n",
       "1                    0.001               -0.455               -0.455    C   \n",
       "2                    0.000               -0.333               -0.333    A   \n",
       "3                    0.000               -0.144               -0.144    C   \n",
       "4                    0.002                0.550                0.550    G   \n",
       "...                    ...                  ...                  ...  ...   \n",
       "27830                0.004                  NaN               -0.397    T   \n",
       "27831                0.197                  NaN                0.509    C   \n",
       "27832                0.021                  NaN                0.439    T   \n",
       "27833                0.000                  NaN                0.000    C   \n",
       "27834                0.000                  NaN                0.000    C   \n",
       "\n",
       "      slr_dnds  upstream  verPhCons_CADD-v1.3 verPhCons_CADD-v1.4  \\\n",
       "0      0.10090         0                  0.0               0.000   \n",
       "1      0.10090         0                  0.0               0.000   \n",
       "2      0.10090         0                  0.0               0.000   \n",
       "3      0.10090         0                  0.0               0.000   \n",
       "4      0.10090         0                  0.0               0.000   \n",
       "...        ...       ...                  ...                 ...   \n",
       "27830  0.00000         0                  NaN               0.000   \n",
       "27831  0.00000         0                  NaN               0.001   \n",
       "27832  0.07122         0                  NaN               0.002   \n",
       "27833  0.12810         0                  NaN               0.009   \n",
       "27834  0.12810         0                  NaN               0.205   \n",
       "\n",
       "      verPhyloP_CADD-v1.3  verPhyloP_CADD-v1.4  \n",
       "0                  -0.395               -0.395  \n",
       "1                   0.301                0.301  \n",
       "2                  -1.268               -1.268  \n",
       "3                  -0.842               -0.842  \n",
       "4                  -0.137               -0.137  \n",
       "...                   ...                  ...  \n",
       "27830                 NaN               -0.325  \n",
       "27831                 NaN                0.186  \n",
       "27832                 NaN                0.439  \n",
       "27833                 NaN                0.509  \n",
       "27834                 NaN                0.509  \n",
       "\n",
       "[27835 rows x 48 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GC -0.00019760009612182456\n",
      "CpG -0.0002719597628887768\n",
      "priPhCons 2.1482277067332688e-19\n",
      "mamPhCons 9.866142392020419e-19\n",
      "verPhCons 8.802311612967998e-19\n",
      "priPhyloP 1.1387414173899128e-18\n",
      "mamPhyloP -9.373178041563204e-20\n",
      "verPhyloP -4.480927534497303e-20\n",
      "GerpN -9.821894011638039e-20\n",
      "GerpS -2.2809728478804073e-20\n",
      "bStatistic 0.0\n"
     ]
    }
   ],
   "source": [
    "for col in [\"GC\", \"CpG\", \"priPhCons\", \"mamPhCons\", \"verPhCons\", \"priPhyloP\", \"mamPhyloP\", \"verPhyloP\", \"GerpN\", \"GerpS\", \"bStatistic\"]:\n",
    "    print(col, (merged[col + '_CADD-v1.3'] - merged[col + '_CADD-v1.4']).sum()/merged.shape[0])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annovar_region</th>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>874714</td>\n",
       "      <td>874714</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>1</td>\n",
       "      <td>905775</td>\n",
       "      <td>905775</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>splicing</td>\n",
       "      <td>AGRN(NM_001305275:exon8:c.1603+10G&gt;A,NM_198576...</td>\n",
       "      <td>1</td>\n",
       "      <td>978847</td>\n",
       "      <td>978847</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>1</td>\n",
       "      <td>980669</td>\n",
       "      <td>980669</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>intronic</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>1</td>\n",
       "      <td>980948</td>\n",
       "      <td>980948</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  annovar_region                                               gene chr  \\\n",
       "0         exonic                                             SAMD11   1   \n",
       "1         exonic                                            PLEKHN1   1   \n",
       "2       splicing  AGRN(NM_001305275:exon8:c.1603+10G>A,NM_198576...   1   \n",
       "3       intronic                                               AGRN   1   \n",
       "4       intronic                                               AGRN   1   \n",
       "\n",
       "    start     end ref alt  \n",
       "0  874714  874714   C   T  \n",
       "1  905775  905775   A   G  \n",
       "2  978847  978847   G   A  \n",
       "3  980669  980669   A   G  \n",
       "4  980948  980948   C   T  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annovar = pd.read_csv('../data/data_nc.ncboost.variant_function', sep='\\t', names=['annovar_region', 'gene', 'chr', 'start', 'end', 'ref', 'alt'])\n",
    "annovar.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "intronic                 27019\n",
       "ncRNA_intronic           15623\n",
       "UTR3                     13839\n",
       "splicing                 10101\n",
       "UTR5                      3521\n",
       "exonic                    2858\n",
       "ncRNA_exonic              1594\n",
       "upstream                   466\n",
       "ncRNA_exonic;splicing      198\n",
       "downstream                 187\n",
       "intergenic                 103\n",
       "UTR5;UTR3                   73\n",
       "exonic;splicing             57\n",
       "ncRNA_splicing              43\n",
       "upstream;downstream         20\n",
       "Name: annovar_region, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annovar.annovar_region.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "INTRONIC            44335\n",
       "DOWNSTREAM          16505\n",
       "UPSTREAM             9658\n",
       "3PRIME_UTR           8784\n",
       "5PRIME_UTR           1602\n",
       "NONCODING_CHANGE     1292\n",
       "INTERGENIC              9\n",
       "SPLICE_SITE             3\n",
       "Name: Consequence, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.Consequence.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "annovar['chr'] = annovar['chr'].apply(lambda x: str(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(75766, 167)"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = annovar.merge(data, left_on=['chr', 'start', 'ref', 'alt'], right_on=['#Chrom', 'Pos', 'Ref', 'Alt'], how='inner')\n",
    "merged.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['annovar_region',\n",
       " 'gene',\n",
       " 'chr',\n",
       " 'start',\n",
       " 'end',\n",
       " 'ref',\n",
       " 'alt',\n",
       " '#Chrom',\n",
       " 'Pos',\n",
       " 'Ref',\n",
       " 'Alt',\n",
       " 'NCBoost_closest_gene_name',\n",
       " 'NCBoost_annovar_genomic_region',\n",
       " 'NCBoost_OMIM_diseases',\n",
       " 'NCBoost_HGMD_DM_public',\n",
       " 'NCBoost_ClinVar',\n",
       " 'NCBoost_Smedley',\n",
       " 'NCBoost_data',\n",
       " 'Type',\n",
       " 'Length',\n",
       " 'AnnoType',\n",
       " 'Consequence',\n",
       " 'ConsScore',\n",
       " 'ConsDetail',\n",
       " 'GC',\n",
       " 'CpG',\n",
       " 'motifECount',\n",
       " 'motifEName',\n",
       " 'motifEHIPos',\n",
       " 'motifEScoreChng',\n",
       " 'oAA',\n",
       " 'nAA',\n",
       " 'GeneID',\n",
       " 'FeatureID',\n",
       " 'GeneName',\n",
       " 'CCDS',\n",
       " 'Intron',\n",
       " 'Exon',\n",
       " 'cDNApos',\n",
       " 'relcDNApos',\n",
       " 'CDSpos',\n",
       " 'relCDSpos',\n",
       " 'protPos',\n",
       " 'relProtPos',\n",
       " 'Domain',\n",
       " 'Dst2Splice',\n",
       " 'Dst2SplType',\n",
       " 'minDistTSS',\n",
       " 'minDistTSE',\n",
       " 'SIFTcat',\n",
       " 'SIFTval',\n",
       " 'PolyPhenCat',\n",
       " 'PolyPhenVal',\n",
       " 'priPhCons',\n",
       " 'mamPhCons',\n",
       " 'verPhCons',\n",
       " 'priPhyloP',\n",
       " 'mamPhyloP',\n",
       " 'verPhyloP',\n",
       " 'bStatistic',\n",
       " 'targetScan',\n",
       " 'mirSVR-Score',\n",
       " 'mirSVR-E',\n",
       " 'mirSVR-Aln',\n",
       " 'cHmmTssA',\n",
       " 'cHmmTssAFlnk',\n",
       " 'cHmmTxFlnk',\n",
       " 'cHmmTx',\n",
       " 'cHmmTxWk',\n",
       " 'cHmmEnhG',\n",
       " 'cHmmEnh',\n",
       " 'cHmmZnfRpts',\n",
       " 'cHmmHet',\n",
       " 'cHmmTssBiv',\n",
       " 'cHmmBivFlnk',\n",
       " 'cHmmEnhBiv',\n",
       " 'cHmmReprPC',\n",
       " 'cHmmReprPCWk',\n",
       " 'cHmmQuies',\n",
       " 'GerpRS',\n",
       " 'GerpRSpval',\n",
       " 'GerpN',\n",
       " 'GerpS',\n",
       " 'TFBS',\n",
       " 'TFBSPeaks',\n",
       " 'TFBSPeaksMax',\n",
       " 'tOverlapMotifs',\n",
       " 'motifDist',\n",
       " 'Segway',\n",
       " 'EncH3K27Ac',\n",
       " 'EncH3K4Me1',\n",
       " 'EncH3K4Me3',\n",
       " 'EncExp',\n",
       " 'EncNucleo',\n",
       " 'EncOCC',\n",
       " 'EncOCCombPVal',\n",
       " 'EncOCDNasePVal',\n",
       " 'EncOCFairePVal',\n",
       " 'EncOCpolIIPVal',\n",
       " 'EncOCctcfPVal',\n",
       " 'EncOCmycPVal',\n",
       " 'EncOCDNaseSig',\n",
       " 'EncOCFaireSig',\n",
       " 'EncOCpolIISig',\n",
       " 'EncOCctcfSig',\n",
       " 'EncOCmycSig',\n",
       " 'Grantham',\n",
       " 'Dist2Mutation',\n",
       " 'Freq100bp',\n",
       " 'Rare100bp',\n",
       " 'Sngl100bp',\n",
       " 'Freq1000bp',\n",
       " 'Rare1000bp',\n",
       " 'Sngl1000bp',\n",
       " 'Freq10000bp',\n",
       " 'Rare10000bp',\n",
       " 'Sngl10000bp',\n",
       " 'dbscSNV-ada_score',\n",
       " 'dbscSNV-rf_score',\n",
       " 'RawScore',\n",
       " 'PHRED',\n",
       " 'Allergy/Immunology/Infectious',\n",
       " 'Audiologic/Otolaryngologic',\n",
       " 'Biochemical',\n",
       " 'Cardiovascular',\n",
       " 'Craniofacial',\n",
       " 'Dental',\n",
       " 'Dermatologic',\n",
       " 'Endocrine',\n",
       " 'Gastrointestinal',\n",
       " 'General',\n",
       " 'Genitourinary',\n",
       " 'Hematologic',\n",
       " 'Musculoskeletal',\n",
       " 'Neurologic',\n",
       " 'Obstetric',\n",
       " 'Oncologic',\n",
       " 'Ophthalmologic',\n",
       " 'Pulmonary',\n",
       " 'Renal',\n",
       " 'Stars',\n",
       " 'allvalid',\n",
       " 'binarized_label',\n",
       " 'chr_pos_ref_alt',\n",
       " 'chr_pos_ref_alt_gene',\n",
       " 'clinpred',\n",
       " 'fathmm_score',\n",
       " 'inClinvar',\n",
       " 'inClinvar1Star',\n",
       " 'inClinvar2Star',\n",
       " 'inTest',\n",
       " 'inVKGLInsertion',\n",
       " 'inheritance',\n",
       " 'isAR',\n",
       " 'isInsertion',\n",
       " 'isPopulation',\n",
       " 'isVKGL_needsFurtherCorrection',\n",
       " 'label',\n",
       " 'max_AF',\n",
       " 'notinTest1',\n",
       " 'oldTestset',\n",
       " 'provean',\n",
       " 'revel',\n",
       " 'sample_weight',\n",
       " 'sift',\n",
       " 'source',\n",
       " 'to_be_deleted']"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>annovar_region</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>874714</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>905775</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>978847</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>splicing</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>980669</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>980948</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>981087</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>981328</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>982356</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>982941</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>984456</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>985266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>985425</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>985434</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>intronic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>985446</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>intronic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>985449</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>intronic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>985797</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>splicing</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>986443</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>987209</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>intronic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>990380</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>1153129</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>exonic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1167621</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>upstream</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1167636</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1219475</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>1226311</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>1226492</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1226630</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>1226912</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>1</td>\n",
       "      <td>1267040</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>1267291</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>1267986</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>1268470</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>1269432</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>1273356</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>1273818</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>1273901</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>1289438</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>1361550</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1475063</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>splicing</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>1573181</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>1956362</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>1956734</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>1</td>\n",
       "      <td>1956870</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>1</td>\n",
       "      <td>1956887</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>1</td>\n",
       "      <td>1957197</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>1958902</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>1958993</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>1958997</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>1959163</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>1</td>\n",
       "      <td>1959186</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>1959757</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>intronic</td>\n",
       "      <td>INTRONIC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #Chrom      Pos Ref Alt Consequence annovar_region Consequence\n",
       "0       1   874714   C   T  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "1       1   905775   A   G  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "2       1   978847   G   A    INTRONIC       splicing    INTRONIC\n",
       "3       1   980669   A   G    INTRONIC       intronic    INTRONIC\n",
       "4       1   980948   C   T    INTRONIC       intronic    INTRONIC\n",
       "5       1   981087   A   G    INTRONIC       intronic    INTRONIC\n",
       "6       1   981328   C   G    INTRONIC       intronic    INTRONIC\n",
       "7       1   982356   G   A    INTRONIC       intronic    INTRONIC\n",
       "8       1   982941   T   C    INTRONIC       intronic    INTRONIC\n",
       "9       1   984456   C   T    INTRONIC       intronic    INTRONIC\n",
       "10      1   985266   C   T    INTRONIC       intronic    INTRONIC\n",
       "11      1   985425   G   A    UPSTREAM       splicing    UPSTREAM\n",
       "12      1   985434   T   A    UPSTREAM       intronic    UPSTREAM\n",
       "13      1   985446   G   T    UPSTREAM       intronic    UPSTREAM\n",
       "14      1   985449   G   A    UPSTREAM       intronic    UPSTREAM\n",
       "15      1   985797   A   G    INTRONIC       splicing    INTRONIC\n",
       "16      1   986443   C   T    INTRONIC       intronic    INTRONIC\n",
       "17      1   987209   C   T    UPSTREAM       intronic    UPSTREAM\n",
       "18      1   990380   C   T  3PRIME_UTR           UTR3  3PRIME_UTR\n",
       "19      1  1153129   C   T    INTRONIC         exonic    INTRONIC\n",
       "20      1  1167621   G   A    UPSTREAM       upstream    UPSTREAM\n",
       "21      1  1167636   G   A    UPSTREAM           UTR5    UPSTREAM\n",
       "22      1  1219475   C   T  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "23      1  1226311   C   T  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "24      1  1226492   G   A  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "25      1  1226630   G   A  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "26      1  1226912   G   A  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "27      1  1267040   T   G  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "28      1  1267291   G   C  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "29      1  1267986   C   G  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "30      1  1268470   G   A  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "31      1  1269432   C   A  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "32      1  1273356   C   A  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "33      1  1273818   T   C  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "34      1  1273901   C   T  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "35      1  1289438   C   T    UPSTREAM         exonic    UPSTREAM\n",
       "36      1  1361550   G   A    UPSTREAM         exonic    UPSTREAM\n",
       "37      1  1475063   C   A  DOWNSTREAM       splicing  DOWNSTREAM\n",
       "38      1  1573181   C   G  DOWNSTREAM         exonic  DOWNSTREAM\n",
       "39      1  1956362   G   A    INTRONIC       intronic    INTRONIC\n",
       "40      1  1956734   C   T    INTRONIC       intronic    INTRONIC\n",
       "41      1  1956870   C   T    INTRONIC       intronic    INTRONIC\n",
       "42      1  1956887   C   G    INTRONIC       intronic    INTRONIC\n",
       "43      1  1957197   G   A    INTRONIC       intronic    INTRONIC\n",
       "44      1  1958902   A   T    INTRONIC       intronic    INTRONIC\n",
       "45      1  1958993   C   T    INTRONIC       intronic    INTRONIC\n",
       "46      1  1958997   G   C    INTRONIC       intronic    INTRONIC\n",
       "47      1  1959163   C   T    INTRONIC       intronic    INTRONIC\n",
       "48      1  1959186   C   T    INTRONIC       intronic    INTRONIC\n",
       "49      1  1959757   G   A    INTRONIC       intronic    INTRONIC"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[['#Chrom', 'Pos', 'Ref', 'Alt', 'Consequence', 'annovar_region', 'Consequence']].head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "def transform(x):\n",
    "    x = x.lower()\n",
    "    if x == '3prime_utr':\n",
    "        return('UTR3')\n",
    "    if x == '5prime_utr':\n",
    "        return('UTR5')\n",
    "    return(x)\n",
    "\n",
    "merged['Consequence_'] = merged.Consequence.apply(lambda x: transform(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged[['#Chrom', 'Pos', 'Ref', 'Alt', 'Consequence', 'annovar_region', 'Consequence', 'Consequence_']].head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.41918538658501175\n",
      "31760 75766\n"
     ]
    }
   ],
   "source": [
    "count = 0\n",
    "total = 0\n",
    "for x in zip(merged['annovar_region'], merged['Consequence_']):\n",
    "    if x[0] == x[1]:\n",
    "        count += 1\n",
    "    total += 1\n",
    "    \n",
    "print(count/total)\n",
    "print(count, total)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "p   b\n",
    "1 2\n",
    "5 6\n",
    "5 10\n",
    "3 0\n",
    "0 10\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 2\n",
      "0 12\n",
      "0 4\n",
      "3 4\n",
      "1 2\n",
      "0 1\n",
      "0 17\n",
      "0 7\n",
      "1 12\n",
      "0 1\n",
      "0 2\n",
      "0 31\n",
      "0 187\n",
      "0 11\n",
      "0 1\n",
      "0 2\n",
      "0 2\n",
      "0 2\n",
      "4 53\n",
      "0 1\n",
      "0 21\n",
      "1 10\n",
      "0 6\n",
      "0 3\n",
      "0 1\n",
      "0 2\n",
      "5 8\n",
      "5 28\n",
      "3 111\n",
      "1 10\n",
      "1 5\n",
      "0 12\n",
      "0 20\n",
      "0 9\n",
      "0 4\n",
      "0 1\n",
      "0 4\n",
      "0 18\n",
      "0 8\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 8\n",
      "0 6\n",
      "0 2\n",
      "0 5\n",
      "0 1\n",
      "0 4\n",
      "0 5\n",
      "0 1\n",
      "0 16\n",
      "0 11\n",
      "0 1\n",
      "0 13\n",
      "0 1\n",
      "0 21\n",
      "0 1\n",
      "0 9\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 6\n",
      "0 2\n",
      "0 4\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 6\n",
      "0 1\n",
      "0 15\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 4\n",
      "0 11\n",
      "0 2\n",
      "0 3\n",
      "0 7\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 2\n",
      "0 20\n",
      "0 1\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 4\n",
      "0 22\n",
      "0 1\n",
      "0 7\n",
      "0 3\n",
      "0 2\n",
      "0 4\n",
      "0 1\n",
      "0 11\n",
      "0 5\n",
      "0 3\n",
      "0 2\n",
      "0 1\n",
      "0 13\n",
      "0 1\n",
      "0 4\n",
      "0 1\n",
      "0 1\n",
      "0 14\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 4\n",
      "0 1\n",
      "0 7\n",
      "0 10\n",
      "0 2\n",
      "0 3\n",
      "0 9\n",
      "0 10\n",
      "0 7\n",
      "0 2\n",
      "0 14\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 6\n",
      "0 2\n",
      "0 4\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 4\n",
      "0 3\n",
      "0 2\n",
      "0 2\n",
      "0 2\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 2\n",
      "0 4\n",
      "0 2\n",
      "0 1\n",
      "0 4\n",
      "0 8\n",
      "0 3\n",
      "0 1\n",
      "0 3\n",
      "0 7\n",
      "0 10\n",
      "0 1\n",
      "0 8\n",
      "0 3\n",
      "0 2\n",
      "0 3\n",
      "0 1\n",
      "0 1\n",
      "0 2\n",
      "0 8\n",
      "0 4\n",
      "0 1\n",
      "0 3\n",
      "0 4\n",
      "0 3\n",
      "0 1\n",
      "0 9\n",
      "0 2\n",
      "0 3\n",
      "0 2\n",
      "0 2\n",
      "0 1\n",
      "0 3\n",
      "1 4\n",
      "0 15\n",
      "0 1\n",
      "9 12\n",
      "3 1\n",
      "0 17\n",
      "18 3\n",
      "0 3\n",
      "0 4\n",
      "0 11\n",
      "0 6\n",
      "0 17\n",
      "0 3\n",
      "0 10\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 1\n",
      "0 14\n",
      "0 2\n",
      "0 1\n",
      "0 17\n",
      "0 76\n",
      "0 22\n",
      "0 5\n",
      "0 78\n",
      "0 8\n",
      "1 7\n",
      "0 2\n",
      "0 6\n",
      "0 4\n",
      "0 42\n",
      "0 1\n",
      "1 10\n",
      "0 3\n",
      "0 1\n",
      "0 1\n",
      "0 66\n",
      "1 36\n",
      "0 3\n",
      "0 1\n",
      "0 2\n",
      "0 8\n",
      "4 2\n",
      "0 4\n",
      "0 10\n",
      "0 7\n",
      "0 1\n",
      "0 10\n",
      "0 1\n",
      "0 3\n",
      "0 44\n",
      "0 6\n",
      "0 8\n",
      "0 4\n",
      "0 1\n",
      "0 1\n",
      "0 7\n",
      "0 3\n",
      "0 15\n",
      "0 15\n",
      "0 3\n",
      "0 1\n",
      "0 2\n",
      "0 25\n",
      "0 13\n",
      "0 3\n",
      "0 2\n",
      "0 98\n",
      "0 6\n",
      "0 2\n",
      "0 1\n",
      "1 13\n",
      "0 1\n",
      "0 1\n",
      "0 18\n",
      "0 11\n",
      "0 1\n",
      "0 3\n",
      "0 1\n",
      "0 2\n",
      "0 6\n",
      "0 16\n",
      "2 1\n",
      "2 5\n",
      "0 1\n",
      "0 1\n",
      "2 6\n",
      "1 39\n",
      "3 1\n",
      "0 16\n",
      "0 12\n",
      "1 7\n",
      "0 9\n",
      "0 5\n",
      "0 1\n",
      "0 31\n",
      "0 2\n",
      "0 2\n",
      "0 6\n",
      "0 2\n",
      "0 9\n",
      "0 3\n",
      "0 5\n",
      "0 25\n",
      "0 2\n",
      "0 28\n",
      "1 4\n",
      "0 2\n",
      "0 4\n",
      "0 11\n",
      "0 1\n",
      "0 2\n",
      "0 1\n",
      "0 20\n",
      "0 1\n",
      "0 4\n",
      "0 12\n",
      "0 13\n",
      "0 1\n",
      "0 10\n",
      "0 18\n",
      "0 2\n",
      "0 1\n",
      "0 17\n",
      "0 3\n",
      "0 9\n",
      "0 7\n",
      "0 3\n",
      "0 2\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 7\n",
      "0 1\n",
      "0 4\n",
      "0 2\n",
      "0 6\n",
      "0 7\n",
      "0 3\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "0 1\n",
      "1 5\n",
      "3 3\n",
      "0 2\n",
      "0 11\n",
      "0 1\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'Benign'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m//anaconda3/envs/py3_vp/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4735\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4736\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mlibindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value_box\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4737\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mIndexError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_box\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_at\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.get_value_at\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.validate_indexer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'str' object cannot be interpreted as an integer",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-109-101f90f50b22>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0mp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0mb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlabel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalue_counts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Benign'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \u001b[0mcount\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m//anaconda3/envs/py3_vp/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1067\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1068\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1070\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m//anaconda3/envs/py3_vp/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4742\u001b[0m                     \u001b[0;32mraise\u001b[0m \u001b[0mInvalidIndexError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4743\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4744\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4745\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# pragma: no cover\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4746\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m//anaconda3/envs/py3_vp/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4728\u001b[0m         \u001b[0mk\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_convert_scalar_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"getitem\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4729\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4730\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtz\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"tz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4731\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4732\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mholds_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_boolean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Benign'"
     ]
    }
   ],
   "source": [
    "count = 0\n",
    "\n",
    "\n",
    "\n",
    "for gene, df in g:\n",
    "    try:\n",
    "        p = df.label.value_counts()['Pathogenic']\n",
    "    except:\n",
    "        p = 0\n",
    "    b = df.label.value_counts()['Benign']\n",
    "    if p != 0:\n",
    "        print(p, b)\n",
    "    count += 1\n",
    "    if count == 1000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (py3_vp)",
   "language": "python",
   "name": "py3_vp"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
